api/v1alpha1/space_types.go

```diff
@@ -0,0 +1,66 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// These are valid status condition reasons of a UserAccount
+const (
+	// Status condition reasons
+	SpaceUnableToCreateNSTemplateSetReason = "UnableToCreateNSTemplateSet"
+	SpaceProvisioningReason                = provisioningReason
+	SpaceProvisionedReason                 = provisionedReason
+	SpaceTerminatingReason                 = terminatingReason
+	SpaceNSTemplateSetUpdateFailedReason   = "NSTemplateSetUpdateFailed"
+)
+
+// SpaceSpec defines the desired state of Space
+type SpaceSpec struct {
+
+	// TargetCluster The cluster in which this Space is provisioned
+	// If not set then the target cluster will be picked automatically
+	// +optional
+	TargetCluster string `json:"targetCluster,omitempty"`
+
+	// TierName is a required property introduced to retain the name of the tier
+	// for which this Space is provisioned
+	TierName string `json:"tierName"`
+}
+
+// SpaceStatus defines the observed state of Space
+type SpaceStatus struct {
+
+	// Conditions is an array of current Space conditions
+	// Supported condition types: ConditionReady
+	// +optional
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// Space is the Schema for the spaces API
```

Comment:
```suggestion
// Space is the Schema for the spaces API
// +k8s:openapi-gen=true
```
Could you please add also the printcolumn annotations?